Userland/Libraries/LibWeb/XHR/XMLHttpRequest.cpp

```diff
@@ -486,10 +509,15 @@ DOM::ExceptionOr<void> XMLHttpRequest::send(Optional<XMLHttpRequestBodyInit> bod
 
     auto request = LoadRequest::create_for_url_on_page(request_url, m_window->page());
     request.set_method(m_method);
-    if (!body_with_type.body.is_empty()) {
-        request.set_body(body_with_type.body);
-        if (!body_with_type.type.is_empty())
-            request.set_header("Content-Type", body_with_type.type);
+    if (body_with_type.has_value()) {
+        body_with_type->body.source().visit(
+            [&](Empty) {},
+            [&](ByteBuffer const& buffer) {
+                request.set_body(buffer);
+            },
+            [&](NonnullRefPtr<FileAPI::Blob> const&) {});
```

Comment:
nit: This can be a bit more compact:
```suggestion
        body_with_type->body.source().visit(
            [&](ByteBuffer const& buffer) { request.set_body(buffer); },
            [](auto&) {});
```

Reply:
Kinda pointless, callback gets populated in the next commit anyway and would need to be split up again

Reply:
We get rid of the `[&](Empty) {}` callback which is nice